compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2020,7 +2020,14 @@ public void visit(BLangRecordLiteral recordLiteral) {
 
         Set<Object> names = new HashSet<>();
         BType type = recordLiteral.type;
-        boolean isOpenRecord = type != null && type.tag == TypeTags.RECORD && !((BRecordType) type).sealed;
+        boolean isRecord = type != null && type.tag == TypeTags.RECORD;
```

Comment:
Not related to this PR, but there shouldn't be a case where the type of a record literal is `null` in code analyzer right?

Reply:
Yeah, ideally should never be null.